packages/db-mongodb/src/updateMany.ts

```diff
@@ -48,6 +49,15 @@ export const updateMany: UpdateMany = async function updateMany(
   transform({ adapter: this, data, fields: collectionConfig.fields, operation: 'write' })
 
   try {
+    if (typeof limit === 'number') {
+      const documentsToUpdate = await Model.find(query, {}, { ...options, limit })
+      if (documentsToUpdate.length === 0) {
+        return null
+      }
+
+      query = { _id: { $in: documentsToUpdate.map((doc) => doc._id) } }
+    }
+
     await Model.updateMany(query, data, options)
```

Comment:
Sadly mongoose does not support specifying a `limit` to `Mode.updateMany`. This should be the fastest way to do it - here we're using `Mode.find` which does support `limit`. The `where` query is essentially offloaded to `Model.find`. 

In the following `Model.updateMany` call, the `where` query will then just be a simple list of IDs